library/src/main/scala/scala/scalajs/js/typedarray/TypedArray.scala

```diff
@@ -44,6 +44,7 @@ trait TypedArray[T, Repr] extends ArrayBufferView with js.Iterable[T] {
   @JSBracketAccess
   def set(index: Int, value: T): Unit = js.native
 
+  // FIXME: Not type-safe. BigInt*Array should accept only BigInt, other TypedArray accept only non-BigInt number.
```

Comment:
Actually, I think this should be fixed *now*, because it was correct before, but the introduction of the `BigInt` typed arrays makes this incorrect. You can make it correct (although more restrictive than strictly necessary) by using `T` as the first type parameter:
```scala
 def set(typedArray: TypedArray[T, _]): Unit = js.native
```
This won't be backward source compatible, but it is backward binary compatible, so it's fine. It deserves to be mentioned in the commit message though, with something along the lines of

    We restrict the argument type of `TypedArray.set` to have the same
    element type as the target typed array, because `BigInt` typed arrays
    and `number` typed arrays are incompatible. This is a backward source
    incompatible change, but it is binary compatible.

Similarly, I think you should restrict the argument of the constructors of the other typed arrays in a similar way, because it is not valid to give them typed arrays of `BigInt`s.

Reply:
Fixed in https://github.com/scala-js/scala-js/pull/3786/commits/9dc3414b54c5e9b4c946902cfc57ad42318e5251 and https://github.com/scala-js/scala-js/pull/3786/commits/7a47c1fbcf1840df72ef595923345421fd983424

Reply:
This `FIXME` and wildcard are left as intended for 0.6.x to avoid binary incompatibility.
It will be fixed in Scala.js 1.x with some breaking change.